### PR TITLE
[Arktos] The scripts for scale-up + workers environment on AWS Ubuntu1804&Ubuntu2004 and scale-out 2x2 + workers environment on AWS Ubuntu 2004 

### PR DESCRIPTION
**Pull request by q131172019**
**What type of PR is this?**
/kind feature
/kind bug

**What this PR does / why we need it**:

This PR is a combined effort of Ying Huang (Sindica),  Hong-Chang, Vinay Kulkarni and Carl Xie (q131172019)

This PR makes the four major changes so that scale-up cluster (Ubuntu 18.04 and Ubuntu 20.04) and scale-out 2 x 2 cluster (Ubuntu 20.04) works  when 1) no cniplugin setting default and 2) cniplugin=flannel and 3) cniplugin=mizar.

1) Arktos script changes necessary to test mizar in arktos-up
2) Add workers in scale-up U1804&U2004 and scale-out U2004
3) Make local scale out have same architecture as kube up
4) Add TP master into mizar droplets in local scale-out

Scale-up environment Ubuntu 18.04 (master + 3 workers)
------------------------------------------------------------------
master: ip-172-31-13-51 (172.31.13.51)
worker1: ip-172-31-11-14 (172.31.11.14)
worker2:ip-172-31-0-171 (172.31.0.171)
worker3: ip-172-31-9-52 (172.31.9.52)

Scale-up environment Ubuntu 20.04 (master + 3 workers)
------------------------------------------------------------------
master: ip-172-31-3-143 (172.31.3.143)
worker1: ip-172-31-9-60 (172.31.9.60)
worker2: ip-172-31-13-125 (172.31.13.125)
worker3:ip-172-31-4-161 (172.31.4.161)

Scale-out 2x2 environment Ubuntu 20.04 (2 TPs X 2 RPs and each RP with 1 worker)
--------------------------------------------------------------------------------
TP1: ip-172-31-4-135 (172.31.4.135)
TP2: ip-172-31-1-189 (172.31.1.189)
RP1: ip-172-31-2-248 (172.31.2.248)
RP1-Worker1 : ip-172-31-8-138 (172.31.8.138)
RP2: ip-172-31-9-41 (172.31.9.41)
RP2-Worker1 : ip-172-31-10-32 (172.31.10.32)

**Which issue(s) this PR fixes**:
Fixes # N/A

**Special notes for your reviewer**:
This PR depends on the efforts of the following PRs
#1230
#1239
#1240
#1265
#1304
#1353
#1354 
#1355

**Does this PR introduce a user-facing change?**:
Test Steps:
***1. Environment1 - No cniplugin setting as default***

****1.1) scale-up standalone on Ubuntu18.04 and Ubuntu20.04****
```
./hack/arktos-up.sh
```

****1.2) scale-up with 3 workers on Ubuntu18.04 and Ubuntu20.04****
Master: 
```
./hack/arktos-up.sh
```

Worker1-3:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from master to worker
API_HOST=<master IP> ./hack/arktos-worker-up.sh
```

***2.  Environment2 - cniplugin=flannel***

****2.1) standalone on Ubuntu18.04 and Ubuntu20.04****
```
CNIPLUGIN=flannel ./hack/arktos-up.sh
```

****2.2) with 3 workers on Ubuntu18.04 and Ubuntu20.04****
Master: 
```
CNIPLUGIN=flannel ./hack/arktos-up.sh
```

Worker1-3:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from master to worker
API_HOST=<master IP> CNIPLUGIN=flannel ./hack/arktos-worker-up.sh
```

**** Test plan: [service support] end-to-end verification of service in scale-up cluster at https://github.com/CentaurusInfra/arktos/issues/1142

****2.3) scale-out standalone on Ubuntu20.04****
Follow up the steps at https://github.com/CentaurusInfra/arktos/blob/master/docs/setup-guide/scale-out-local-dev-setup.md
TP1:
```
SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.0.0.0/16
make clean; ./hack/arktos-up-scale-out-poc.sh
```

TP2:
```
SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.1.0.0/16
make clean; ./hack/arktos-up-scale-out-poc.sh
```

RP1:
```
IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.244.0.0/16  ./hack/arktos-up-scale-out-poc.sh
```

RP2:
```
IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.245.0.0/16  ./hack/arktos-up-scale-out-poc.sh
```

Patch RP1 and RP2:
```
1. stop the source/dest check (on AWS console, using EC2 instance Action menu, choosing Networking | Change source-destination check);
2. manually add relevant routing entries of each node, so that each routing table is complete for all nodes of whole cluster across RPs, e.g. (assuming pod cidr of rp0 is 10.244.0.0/16, rp1 10.245.0.0/16)
on RP1,

   sudo ip r add 10.245.0.0/24 via [RP2-IP]
on RP2

   sudo ip r add 10.244.0.0/24 via [RP1-IP]
```

**** Test plan: [service support] end-to-end verification of service in scale-out cluster at https://github.com/CentaurusInfra/arktos/issues/1143

****2.4) scale-out 2x2 - each RP with 1 worker on Ubuntu20.04****
Follow up the steps at https://github.com/CentaurusInfra/arktos/blob/master/docs/setup-guide/scale-out-local-dev-setup.md
TP1:
```
SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.0.0.0/16
./hack/arktos-up-scale-out-poc.sh
```

TP2:
```
SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.1.0.0/16
./hack/arktos-up-scale-out-poc.sh
```

RP1:
```
IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.244.0.0/16  ./hack/arktos-up-scale-out-poc.sh
```

RP2:
```
IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.245.0.0/16  ./hack/arktos-up-scale-out-poc.sh

```

Patch RP1 and RP2:
```
1. stop the source/dest check (on AWS console, using EC2 instance Action menu, choosing Networking | Change source-destination check);
2. manually add relevant routing entries of each node, so that each routing table is complete for all nodes of whole cluster across RPs, e.g. (assuming pod cidr of rp0 is 10.244.0.0/16, rp1 10.245.0.0/16)
on RP1,

   sudo ip r add 10.245.0.0/24 via [RP2-IP]
on RP2

   sudo ip r add 10.244.0.0/24 via [RP1-IP]
```

RP1-Worker1:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from RP1 master to worker
IS_SCALE_OUT=true API_HOST=<RP1-IP> API_TENANT_SERVER=<TP1-IP, TP2-IP> CNIPLUGIN=flannel ./hack/arktos-worker-up.sh
```

RP2-Worker1:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from RP2 master to worker
IS_SCALE_OUT=true API_HOST=<RP2-IP> API_TENANT_SERVER=<TP1-IP, TP2-IP> CNIPLUGIN=flannel ./hack/arktos-worker-up.sh
```

Patch RP1-Worker1 and RP2-Worker1:
```
1. stop the source/dest check (on AWS console, using EC2 instance Action menu, choosing Networking | Change source-destination check);
```
**** Test plan: [service support] end-to-end verification of service in scale-out cluster at https://github.com/CentaurusInfra/arktos/issues/1143

***3. Environment3 - cniplugin=mizar on Ubuntu18.04 and Ubuntu20.04***

**** Mizar test plan: https://github.com/CentaurusInfra/arktos/wiki/Mizar-Arktos-Integration-Release-2022-0130-Test-Plan

****3.1) standalone on Ubuntu18.04 and Ubuntu20.04****
```
CNIPLUGIN=mizar ./hack/arktos-up.sh
```

****3.2) with 3 workers on Ubuntu18.04 and Ubuntu20.04****

Master: 
```
CNIPLUGIN=mizar ./hack/arktos-up.sh
```

Worker1-3:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from master to worker
API_HOST=<master IP> CNIPLUGIN=mizar ./hack/arktos-worker-up.sh
```

****3.3) scale-out standalone on Ubuntu20.04****
Follow up the steps at https://github.com/CentaurusInfra/arktos/blob/master/docs/setup-guide/scale-out-local-dev-setup.md
TP1:
```
CNIPLUGIN=mizar SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.0.0.0/16 ./hack/arktos-up-scale-out-poc.sh
```

TP2:
```
CNIPLUGIN=mizar SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.1.0.0/16 ./hack/arktos-up-scale-out-poc.sh
```

RP1:
```
CNIPLUGIN=mizar IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.244.0.0/16  ./hack/arktos-up-scale-out-poc.sh
```

RP2:
```
CNIPLUGIN=mizar IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.245.0.0/16  ./hack/arktos-up-scale-out-poc.sh
```

Patch RP1 and RP2:
```
1. stop the source/dest check (on AWS console, using EC2 instance Action menu, choosing Networking | Change source-destination check);
2. manually add relevant routing entries of each node, so that each routing table is complete for all nodes of whole cluster across RPs, e.g. (assuming pod cidr of rp0 is 10.244.0.0/16, rp1 10.245.0.0/16)
on RP1,

   sudo ip r add 10.245.0.0/24 via [RP2-IP]
on RP2

   sudo ip r add 10.244.0.0/24 via [RP1-IP]
```

****3.4) scale-out 2x2 - each RP with 1 worker on Ubuntu20.04****
Follow up the steps at https://github.com/CentaurusInfra/arktos/blob/master/docs/setup-guide/scale-out-local-dev-setup.md
TP1:
```
CNIPLUGIN=mizar SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.0.0.0/16
./hack/arktos-up-scale-out-poc.sh
```

TP2:
```
CNIPLUGIN=mizar SCALE_OUT_TP_ENABLE_DAEMONSET=true IS_RESOURCE_PARTITION=false RESOURCE_SERVER=<RP1-IP, RP2-IP> TENANT_PARTITION_SERVICE_SUBNET=10.1.0.0/16
./hack/arktos-up-scale-out-poc.sh
```

RP1:
```
CNIPLUGIN=mizar IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.244.0.0/16  ./hack/arktos-up-scale-out-poc.sh
```

RP2:
```
CNIPLUGIN=mizar IS_RESOURCE_PARTITION=true TENANT_SERVER=<TP1-IP, TP2-IP> RESOURCE_PARTITION_POD_CIDR=10.245.0.0/16  ./hack/arktos-up-scale-out-poc.sh

```

Patch RP1 and RP2:
```
1. stop the source/dest check (on AWS console, using EC2 instance Action menu, choosing Networking | Change source-destination check);
2. manually add relevant routing entries of each node, so that each routing table is complete for all nodes of whole cluster across RPs, e.g. (assuming pod cidr of rp0 is 10.244.0.0/16, rp1 10.245.0.0/16)
on RP1,

   sudo ip r add 10.245.0.0/24 via [RP2-IP]
on RP2

   sudo ip r add 10.244.0.0/24 via [RP1-IP]
```

RP1-Worker1:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from RP1 master to worker
CNIPLUGIN=mizar IS_SCALE_OUT=true API_HOST=<RP1-IP> API_TENANT_SERVER=<TP1-IP, TP2-IP> CNIPLUGIN=flannel ./hack/arktos-worker-up.sh
```

RP2-Worker1:
```
sudo mkdir -p /var/run/kubernetes
copy /var/run/kubernetes/client-ca.crt from RP2 master to worker
CNIPLUGIN=mizar IS_SCALE_OUT=true API_HOST=<RP2-IP> API_TENANT_SERVER=<TP1-IP, TP2-IP> CNIPLUGIN=flannel ./hack/arktos-worker-up.sh
```

Patch RP1-Worker1 and RP2-Worker1:
```
1. stop the source/dest check (on AWS console, using EC2 instance Action menu, choosing Networking | Change source-destination check);
```

### CODE REVIEW

File: hack/arktos-cni.rc [hunk 1]
```diff
@@ -33,7 +33,7 @@ setup_cni_conf() {
 
     echo "copying ${conf_source_file} to ${conf_target_file}..."
     local cni_conf_dir=$(dirname ${conf_target_file})
-    local backup_dir="$(dirname ${cni_conf_dir})/$(basename ${cni_conf_dir})_"$(date -d "today" +"%Y%m%d%H%M")
+    local backup_dir="/var/backups/$(basename ${cni_conf_dir})_"$(date -d "today" +"%Y%m%d%H%M")
```
Comment (sonyafenge) [inline hack/arktos-cni.rc:36]: what is the reason that this changed to hardcode?

File: hack/arktos-cni.rc [hunk 2]
```diff
@@ -66,6 +66,12 @@ install_cni_bridge() {
     echo "done with bridge cni plugin installation"
 }
 
+cleanup_cni_mizar() {
+    sudo rm -rf /opt/cni/bin/
+    [ ! -d "${cni_conf_dir}" ] && sudo mkdir -p "${cni_conf_dir}"
+    [ ! -d "${cni_bin_dir}" ] && sudo mkdir -p "${cni_bin_dir}"
+    sudo find /etc/cni/net.d/ -type f -not -name '10-mizarcni.conf' -delete
```
Comment (sonyafenge) [inline hack/arktos-cni.rc:73]: Does /opt/cni/bin/ = ${cni_bin_dir}? can we use var to replace hardcode here?

File: hack/arktos-up-scale-out-poc.sh [hunk 3]
```diff
@@ -681,6 +683,49 @@ if [ "${IS_RESOURCE_PARTITION}" == "true" ]; then
...
+  if [[ "${CNIPLUGIN}" == "mizar" ]]; then
+    ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" create configmap system-source --namespace=kube-system --from-literal=name=arktos --from-literal=company=futurewei
+
+    # Creating mizar crds
+    echo "Creating mizar crds ......."
+    cp "${KUBE_ROOT}/third_party/mizar/mizar-crds.yaml" mizar-crds.yaml
+    ${KUBECTL} apply -f mizar-crds.yaml
+    rm mizar-crds.yaml
+
+    if [[ "${IS_SECONDARY_TP}" == "false" ]]; then
```
Comment (sonyafenge) [inline hack/arktos-up-scale-out-poc.sh:696]: when IS_SECONDARY_TP == true? what is the purpose for this condition?
How about 3rd, 4th... TP?

File: hack/arktos-up.sh [hunk 4]
```diff
@@ -554,6 +559,13 @@ if [[ "${START_MODE}" != "nokubelet" ]]; then
     esac
 fi
 
+# Applying mizar cni
+if [[ "${CNIPLUGIN}" == "mizar" ]]; then
+  ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" create configmap system-source --namespace=kube-system --from-literal=name=arktos --from-literal=company=futurewei
+  # ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" apply -f ${KUBE_ROOT}/hack/testdata/mizar/deploy.mizar.next.yaml 
+  ${KUBECTL} --kubeconfig="${CERT_DIR}/admin.kubeconfig" apply -f https://raw.githubusercontent.com/CentaurusInfra/mizar/dev-next/etc/deploy/deploy.mizar.dev.yaml
```
Comment (sonyafenge) [inline hack/arktos-up.sh:566]: suggest to download first and then apply. directly apply may have unstable risk

File: hack/arktos-worker-up.sh [hunk 5]
```diff
@@ -17,27 +17,90 @@
 set -e
 die() { echo "$*" 1>&2 ; exit 1; }
 
-if [[ ! -n "${KUBELET_IP}" ]]; then
-    die "KUBELET_IP env var not set"
+echo "DBG: Flannel CNI plugin will be installed AFTER cluster is up"
+[ "${CNIPLUGIN}" == "flannel" ] && ARKTOS_NO_CNI_PREINSTALLED="y"
+
+OPT_CNI_BIN=/opt/cni/bin
+ETC_CNI_NET_DIR=/etc/cni/net.d
```
Comment (sonyafenge) [inline hack/arktos-worker-up.sh:24]: looks like same value, different var name in different script, suggest to use same var name. 
in [arktos-cni.rc](https://github.com/CentaurusInfra/arktos/pull/1382/files#diff-64c7b42f9b8968b0eaba4bc41373285edf52e57c6d896f0e158969321407f8a1):
```
 cni_conf_dir=${CNI_CONF_DIR:-"/etc/cni/net.d"}
cni_bin_dir=${CNI_BIN_DIR:-"/opt/cni/bin"},
```

File: hack/arktos-worker-up.sh [hunk 6]
```diff
@@ -47,29 +110,141 @@ if [[ ! -s "${KUBELET_CLIENTCA}" ]]; then
...
+
+if [[ -n ${FEATURE_GATES} ]]; then
+  echo "featureGates:"
+  # Convert from foo=true,bar=false to
+  #   foo: true
+  #   bar: false
+  for gate in $(echo "${FEATURE_GATES}" | tr ',' ' '); do
+    echo "${gate}" | sed -e 's/\(.*\)=\(.*\)/  \1: \2/'
+  done
+fi >>/tmp/kube-proxy.yaml
```
Comment (sonyafenge) [inline hack/arktos-worker-up.sh:243]: didn't see where to remove this /tmp/kube-proxy.yaml. this suppose to be deleted once finished usage.

File: hack/lib/common-var-init.sh [hunk 7]
```diff
@@ -109,6 +109,7 @@ EXTERNAL_CLOUD_PROVIDER_BINARY=${EXTERNAL_CLOUD_PROVIDER_BINARY:-""}
 CLOUD_PROVIDER=${CLOUD_PROVIDER:-""}
 CLOUD_CONFIG=${CLOUD_CONFIG:-""}
 FEATURE_GATES=${FEATURE_GATES:-"AllAlpha=false"}
+FEATURE_GATES_COMMON_BASE=${FEATURE_GATES_COMMON_BASE:-"${FEATURE_GATES},WorkloadInfoDefaulting=true,QPSDoubleGCController=true,QPSDoubleRSController=true"}
```
Comment (sonyafenge) [inline hack/lib/common-var-init.sh:112]: what is the reason a new env for FEATURE_GATES need be involved.

File: hack/arktos-cni.rc [hunk 8]
```diff
@@ -158,6 +164,25 @@ install_arktos_containerd() {
...
+      echo "installing arktos containerd..."
+      sudo systemctl stop containerd
+      sudo mv /usr/bin/containerd /usr/bin/containerd.bak
+      sudo cp $HOME/arktos-containerd/containerd /usr/bin/
+      sudo systemctl start containerd
+      sudo systemctl restart docker
+    else
+      echo "arktos containerd found..."
+    fi
+}
```
Comment (yb01) [inline hack/arktos-cni.rc:185]: in kube-up env, we won't need the mizar specific containerd anymore, so this code is not needed anymore.

File: hack/arktos-up-scale-out-poc.sh [hunk 9]
```diff
@@ -114,17 +116,21 @@ if [ "${CLOUD_PROVIDER}" == "openstack" ]; then
...
-FEATURE_GATES="${FEATURE_GATES},WorkloadInfoDefaulting=true,QPSDoubleGCController=true,QPSDoubleRSController=true"
 
 # check for network service support flags
 if [ -z ${DISABLE_NETWORK_SERVICE_SUPPORT} ]; then # when enabled
   # kubelet enforces per-network DNS ip in pod
   FEATURE_GATES="${FEATURE_GATES},MandatoryArktosNetwork=true"
   # tenant controller automatically creates a default network resource for new tenant
-  ARKTOS_NETWORK_TEMPLATE="${KUBE_ROOT}/hack/testdata/default-flat-network.tmpl"
+  if [ "${CNIPLUGIN}" == "mizar" ]; then
+    ARKTOS_NETWORK_TEMPLATE="${KUBE_ROOT}/hack/runtime/default_mizar_network.json"
```
Comment (yb01) [inline hack/arktos-up-scale-out-poc.sh:130]: do we still need this template file? with the recent code changes to construct the mizar network object in the mizar controller code ?

File: hack/arktos-up-scale-out-poc.sh [hunk 10]
```diff
@@ -681,6 +683,49 @@ if [ "${IS_RESOURCE_PARTITION}" == "true" ]; then
...
+
+    # Deploying mizar daemon to tp master
+    echo "Deploying mizar daemon to TP master ......."
+    cp "${KUBE_ROOT}/third_party/mizar/mizar-daemon-tpmaster.yaml" mizar-daemon-tpmaster.yaml
+    sed -i -e "s@{{network_provider_version}}@${MIZAR_VERSION}@g" mizar-daemon-tpmaster.yaml
+    sed -i -e "s@{{tp_master_name}}@${API_HOST}@g" mizar-daemon-tpmaster.yaml
+    ${KUBECTL} apply -f mizar-daemon-tpmaster.yaml
+    rm mizar-daemon-tpmaster.yaml
+
+    # Place mizar operator
```
Comment (yb01) [inline hack/arktos-up-scale-out-poc.sh:713]: mizar. operator is on each TP only. so. this should have a flat to not deploy it on RPs.

File: hack/arktos-up-scale-out-poc.sh (hunk 3 above)
Comment (Sindica) [reply hack/arktos-up-scale-out-poc.sh:696]: > when IS_SECONDARY_TP == true? what is the purpose for this condition? How about 3rd, 4th... TP?

Aside from primary TP, which will have mizar daemonset deployed, all other TPs are secondardy TP.

File: hack/arktos-up.sh (hunk 4 above)
Comment (Sindica) [reply hack/arktos-up.sh:566]: > suggest to download first and then apply. directly apply may have unstable risk

The apply option should make sure download whole file and then apply together. This should not be a problem. However, we need to make sure using stable mizar release for our release.

File: hack/arktos-worker-up.sh (hunk 6 above)
Comment (Sindica) [reply hack/arktos-worker-up.sh:243]: > didn't see where to remove this /tmp/kube-proxy.yaml. this suppose to be deleted once finished usage.

This file in in tmp folder, no need to worry about removing. Keep it there for debugging purpose.

File: hack/arktos-up-scale-out-poc.sh (hunk 9 above)
Comment (Sindica) [reply hack/arktos-up-scale-out-poc.sh:130]: > do we still need this template file? with the recent code changes to construct the mizar network object in the mizar controller code ?

This template file is used by arktos network controller, not mizar VPC/subnet we just introduced.

File: hack/arktos-up-scale-out-poc.sh (hunk 10 above)
Comment (Sindica) [reply hack/arktos-up-scale-out-poc.sh:713]: The outter if statement is for TP only (line 685)

File: hack/arktos-worker-up.sh (hunk 6 above)
Comment (sonyafenge) [reply hack/arktos-worker-up.sh:243]: For cleanup purpose, I suggest we remove all files dynamic generated on dev machine to ensure next start will not be impacted by existing files.

File: hack/arktos-up.sh (hunk 4 above)
Comment (sonyafenge) [reply hack/arktos-up.sh:566]: sounds good.

File: hack/arktos-cni.rc (hunk 8 above)
Comment (q131172019) [reply hack/arktos-cni.rc:185]: In local env, containerd is used. Please see line 250 below in same file.

File: hack/lib/common-var-init.sh (hunk 7 above)
Comment (q131172019) [reply hack/lib/common-var-init.sh:112]: After conversation with sonyafeng, the explanation is given.

File: hack/arktos-cni.rc (hunk 1 above)
Comment (q131172019) [reply hack/arktos-cni.rc:36]: The backup dir should be different from coni conf dir. But /var/backup should be replaced by local variable.

File: hack/arktos-cni.rc (hunk 2 above)
Comment (q131172019) [reply hack/arktos-cni.rc:73]: Yes, use var to replace hardcode.

File: hack/arktos-worker-up.sh (hunk 5 above)
Comment (q131172019) [reply hack/arktos-worker-up.sh:24]: Thanks for your catch-up. use same name defined in hack/arktos-cni.rc.

cni_conf_dir=${CNI_CONF_DIR:-"/etc/cni/net.d"} 
cni_bin_dir=${CNI_BIN_DIR:-"/opt/cni/bin"}

File: hack/arktos-worker-up.sh (hunk 6 above)
Comment (q131172019) [reply hack/arktos-worker-up.sh:243]: line 255 "cat <<EOF > /tmp/kube-proxy.yaml" is similar to clean old contents and fill with new contents.

**Review by sonyafenge — APPROVED** [review]
/LGTM

**Review by yb01 — APPROVED** [review]
/lgtm

**Pull request merged**